src/index.js

```diff
@@ -995,17 +1004,32 @@ module.exports = {
                   : 'application/xml';
               }
 
-              return bubbleError(() =>
-                expect(
-                  makeAttachedDOMNodeList(subject.childNodes, contentType),
-                  'to satisfy',
-                  value.children
-                )
+              const subjectChildren = makeAttachedDOMNodeList(
+                subject.childNodes,
+                contentType
               );
+
+              return bubbleError(() => {
+                if (expect.findTypeOf(value.children).is('expect.it')) {
+                  return value.children(subjectChildren, expect.context);
```

Comment:
Hmm, are you sure this one is necessary? Seems like `<object> to satisfy <object>` already handles this? https://github.com/unexpectedjs/unexpected/blob/master/lib/assertions.js#L1821

If it doesn't maybe we should fix it there.

I don't remember the specificity computations by heart, but it could also be covered by this one: https://github.com/unexpectedjs/unexpected/blob/a7f5e05dc19f0df2e1cb49362c1641e89994acc3/lib/assertions.js#L1364-L1367

Reply:
You are right :) the way deferral is done in U-dom things seem to work, so I've removed that part of the changes but left the test cases in so it doesn't regress.